Riot/Managers/PushRulesUpdater/PushRulesUpdater.swift

```diff
@@ -0,0 +1,81 @@
+// 
+// Copyright 2023 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Combine
+
+final class PushRulesUpdater {
+    private var cancellables: Set<AnyCancellable> = .init()
+    private var rules: [NotificationPushRuleType] = []
+    private let notificationSettingsService: NotificationSettingsServiceType
+    private let didCompleteUpdateSubject: PassthroughSubject<Void, Never> = .init()
+    
+    var didCompleteUpdate: AnyPublisher<Void, Never> {
+        didCompleteUpdateSubject.eraseToAnyPublisher()
+    }
+    
+    init(notificationSettingsService: NotificationSettingsServiceType, needsCheck: AnyPublisher<Void, Never>) {
+        self.notificationSettingsService = notificationSettingsService
+        
+        notificationSettingsService
+            .rulesPublisher
+            .weakAssign(to: \.rules, on: self)
```

Comment:
I'm not sure I understand this, is is the same as: ?

```
.sink { [weak self] value in 
    self?.rules = value
}
```

cause if it is I definitely find mine a lot simpler to understand

Reply:
Yeah, the code is the same.
I think assign is handy every time you need just to assign a value to a key path.
Unfortunately it looks like Combine doesn't have a "weak" variant of [this](https://developer.apple.com/documentation/combine/just/assign(to:on:)).

Reply:
Okay, fair enough 👍